### PR TITLE
Modinreach 327

### PR DESCRIPTION
**Pull request by nirmalr-shah**
[MODINREACH-327](https://issues.folio.org/browse/MODINREACH-327)

## Purpose

INN-Reach: Item's "in-transit" should be "Available" in INN-Reach if there are no open requests on the item.

## Approach
As per my analysis, When the item is checked in and put in-transit, an automatic update triggers ongoing contribution, which internally prepares/updates the item object and contributes to the service catalog. But here in the contribution process, it confirms whether in-transit item is requested or not through just counts of request(s).

And, the check implemented here should check only open requests which are having status "Open - Awaiting pickup" or "Open - Not yet filled" or "Open - In transit" or "Open - Awaiting delivery", but it also considers requests which are having status Closed - Filled, Closed - Cancelled, Closed - Unfilled, Closed - Pickup expired  

Due to this item is considered as requested even with closed requests, and Item's updated contribution status shows as UNAVAILABLE on the service catalog ([https://d2irm.iii.com/)](https://d2irm.iii.com/)*_)

Even on the Inventory Item Page, which shows the number of requests for the item that only shows request-count which with any of "Open - Awaiting pickup", "Open - Not yet filled", "Open - In transit", "Open - Awaiting delivery".

**Finally, decided to consider only “Open - …” requests.**

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/impl/ContributionValidationServiceImpl.java [hunk 1]
```diff
@@ -266,7 +263,7 @@ private boolean isItemAvailableForContribution(Item inventoryItem,
   }
 
   private boolean isItemRequested(Item inventoryItem) {
-    var itemRequests = circulationClient.queryRequestsByItemId(inventoryItem.getId());
+    var itemRequests = circulationClient.queryRequestsByItemIdAndStatus(inventoryItem.getId(),1);
```
Comment (Arghya22) [inline src/main/java/org/folio/innreach/domain/service/impl/ContributionValidationServiceImpl.java:266]: please add an optional or null check and throw appropriate error if fails

**Review by arin-epm — APPROVED** [review]

**Comment by nirmalr-shah** [comment]
attaching evidence of attaching all unit test case successful execution screenshot
image

**Comment by nirmalr-shah** [comment]
Herewith attaching the recording of testing after the fix.
Item now becomes available state if it checked in from a different service point than its effective location with one open loan.

https://epam-my.sharepoint.com/:v:/r/personal/nirmal_shah_epam_com/Documents/chrome_9XaSBbKgjC.mp4?csf=1&web=1&e=WH25u0

**Comment by nirmalr-shah** [comment]
https://epam-my.sharepoint.com/:v:/r/personal/nirmal_shah_epam_com/Documents/chrome_9XaSBbKgjC.mp4?csf=1&web=1&e=WH25u0

File: src/main/java/org/folio/innreach/domain/service/impl/ContributionValidationServiceImpl.java (hunk 1 above)
Comment (nirmalr-shah) [reply src/main/java/org/folio/innreach/domain/service/impl/ContributionValidationServiceImpl.java:266]: main method from which this **_isItemRequested_** is having try-catch block for error handling!!

Also, here **_queryRequestsByItemIdAndStatus_** is returning a list.
Generally, our intention is to put a limited mechanism for library method return types where there needed to be a clear way to represent "no result", and when it comes to lists (and collections in general), there is already a clear way to represent "no result", and that is an empty collection.

**Review by Arghya22 — APPROVED** [review]

**Pull request merged**